include/git2/deprecated.h

```diff
@@ -41,6 +41,13 @@
  */
 #ifndef GIT_DEPRECATE_HARD
 
+/*
+ * The credential structures are now opaque by default, and their
+ * definition has moved into the `sys/credential.h` header; include
+ * them here for backward compatibility.
+ */
+#include "sys/credential.h"
```

Comment:
I thought of adding these headers to "include/git2/credential.h", but this is definitely the better location to have it.